packages/composites/anchor/src/endpoint/price/beth.ts

```diff
@@ -1,57 +1,55 @@
 import { ethers } from 'ethers'
 import { PriceExecute } from '.'
-import BigNumber from 'bignumber.js'
 import { Config } from '../../config'
+import { throwErrorForInvalidResult } from '../../utils'
 import { anchorVaultAbi, curvePoolAbi } from './abi'
 
 export const FROM = 'BETH'
-export const INTERMEDIARY_TOKEN_DECIMALS = 8
 export const INTERMEDIARY_TOKEN = 'ETH'
 
-export const execute: PriceExecute = async (input, _, config, taAdapterResponse) => {
+export const execute: PriceExecute = async (input, _, config, usdPerEth) => {
   const rpcUrl = config.rpcUrl
   const provider = new ethers.providers.JsonRpcProvider(rpcUrl)
-  const stEthPerBEth = await getStEthBEthExchangeRate(config, provider)
-  const ethPerStEth = await getStETHExchangeRate(config, provider)
-  const usdPerEth = taAdapterResponse.data.result
-  const result = new BigNumber(usdPerEth)
-    .multipliedBy(ethPerStEth)
-    .multipliedBy(stEthPerBEth)
-    .toNumber()
-  return {
-    jobRunID: input.id,
-    statusCode: 200,
-    result,
-    data: {
-      result,
-    },
-  }
+  const stEthPerBEth = await getStEthBEthExchangeRate(input.id, config, provider)
+  const stEthPerETH = await getStETHExchangeRate(input.id, config, provider)
+  // result = (USD / ETH) * (stETH / bETH) * (ETH / stETH) = USD / bETH
```

Comment:
I'd prefer to see this kind of formula documenting in the JSdoc style at the top of the function.
That way it gets picked up by VSCode Intellisense.

Reply:
+1 for JSDoc documentation